### PR TITLE
Replacing deletion with archive operation for Content page

### PR DESCRIPTION
**Pull request by tonyzhao6**
Reviewer: @sidravi1 
Estimate: ~15 minutes

---

## Ticket

Fixes: N/A

### Goal

1. Replace the "delete" button with the "archive" button on the Content page.
2. When the user clicks the "archive" button, a new message pops up informing the user about the archive operation.
3. If the user continues to archive, then the content no longer displays on the Content page.
4. The archived content is still available on the backend by using the query parameter "exclude_archived".

### Changes

[Example demonstration](https://www.loom.com/share/05d5117f833d446e9e17ec83f3519e5a?sid=415cebee-d59c-4109-b0f5-1d7a687bb98e)

1. Changed admin_app Content page functionalities.
2. Updated content backend.
3. Updated tests and added new tests for archiving operations.

## How has this been tested?

1. With unit tests for the archive endpoints.

## Checklist

Fill with `x` for completed. 

- [X] My code follows the style guidelines of this project
- [X] I have reviewed my own code to ensure good quality
- [X] I have tested the functionality of my code to ensure it works as intended
- [X] I have resolved merge conflicts

### CODE REVIEW

**Review by sidravi1 — APPROVED** [review]
lgtm! Thanks @tonyzhao6 
Just left some minor comments and to delete the commented out old code.

File: admin_app/src/components/ContentCard.tsx [hunk 1]
```diff
@@ -127,10 +137,11 @@ const ContentCard = ({
             size="small"
```
Comment (sidravi1) [inline admin_app/src/components/ContentCard.tsx:137]: Looked nicer with "medium" with the new icon. But i'll let you decide!

File: admin_app/src/components/ContentCard.tsx [hunk 2]
```diff
@@ -1,9 +1,11 @@
 import {
   ContentViewModal,
-  DeleteContentModal,
+  ArchiveContentModal,
+  // DeleteContentModal,
 } from "@/components/ContentModal";
 import { appColors, appStyles, sizes } from "@/utils";
-import { Delete, Edit } from "@mui/icons-material";
+// import { Delete, Edit } from "@mui/icons-material";
+import { Archive, Edit } from "@mui/icons-material";
```
[the remark this answers is not in the document]
Comment (tonyzhao6) [reply admin_app/src/components/ContentCard.tsx:8]: Yup, can use this icon instead and will delete the commented out old code.

File: admin_app/src/components/ContentCard.tsx (hunk 1 above)
Comment (tonyzhao6) [reply admin_app/src/components/ContentCard.tsx:137]: yes, medium looks better

**Pull request merged**